sdk/cosmos/src/errors.rs

```diff
@@ -1,17 +1,22 @@
 /// An error having to do with Cosmos.
+#[allow(missing_docs)]
 #[non_exhaustive]
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
     /// An error as defined in the `azure_core` crate
     #[error(transparent)]
-    AzureCoreError(#[from] azure_core::Error),
+    Core(#[from] azure_core::Error),
     /// An error related to parsing
     #[error(transparent)]
-    ParsingError(ParsingError),
-    // TODO forward this to azure_core::Error
-    #[allow(missing_docs)]
+    ParsingError(#[from] ParsingError),
+    #[error("http error: {0}")]
+    CoreHttpError(#[from] azure_core::HttpError),
+    #[error("stream error: {0}")]
+    StreamError(#[from] azure_core::StreamError),
```

Comment:
I would not like to do this. See the discussion in https://github.com/Azure/azure-sdk-for-rust/pull/306#discussion_r654666302. The idea was that `cosmos::Error` would only contain errors that can _only_ happen in the cosmos SDK and not in any other SDK (plus a final catch all `AzureCoreError`  - or `Core` if you prefer). What code (whether in core or in cosmos) triggered the error is a concern of the `source` chain for the error that is produced and not something we should encode in the type.

Reply:
These errors are happening in the cosmos SDK. This reverts the removal of them in https://github.com/Azure/azure-sdk-for-rust/pull/306.  https://github.com/Azure/azure-sdk-for-rust/pull/306 has a bunch of great changes, but I think this should be a separate PR.